modules/optimizing-by-encapsulation.adoc

```diff
@@ -0,0 +1,38 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/optimizing-cpu-usage
+
+[id="optimizing-cpu-usage_{context}"]
+= Encapsulating mount namespaces
+
+Mount namespaces are used to isolate mount points so that processes in different namespaces cannot view each others' files. Encapsulation is the process of moving mount namespaces to an alternative location, known to the owner but hidden from intrusive visitors.
+
+The host operating system uses `systemd` to constantly scan all mount namespaces: both the standard Linux mounts and the numerous mounts that Kubernetes uses to operate. The current implementation of Kubelet and CRI-O both use the top-level namespace for all container and Kubelet mountpoints. However, encapsulating these container-specific mountpoints in a private namespace reduces `systemd` overhead with no difference in functionality. Using a separate mount namespace for both CRI-O and Kubelet can encapsulate container-specific mounts from any `systemd` or other host OS interaction.
+
+This ability to potentially achieve major CPU optimization is now available to all {product-title} administrators. Encapsulation can also improve security, by storing Kubernetes-specific mount points in a location safe from inspection by unprivileged users.
+
+The following diagrams illustrate a Kubernetes installation before and after encapsulation. Both scenarios show example containers which have mount propagation settings of bidirectional, host-to-container, and none.
+
+image::before-k8s-mount-propagation.png[Before encapsulation]
+
+Here we see `systemd`, host OS processes, `kubelet`, and the container runtime sharing a single mount namespace.
+
+* `systemd`, host OS processes, `kubelet`, and the container runtime each have access to and visibility of all mount points.
+
+* A mount originating in Container 1, such as `/run/a` is visible to systemd, host OS processes, Kubelet, CRI-O, and other containers with host-to-container or bidirectional mount propagation configured.
+
+The following diagram illustrates the system state after encapsulation.
+
+image::after-k8s-mount-propagation.png[After encapsulation]
```

Comment:
Replace with the new https://github.com/redhataccess/documentation-svg-assets/blob/master/for-web/OpenShift/290_OpenShift/290_OpenShift_encapsulating_mount_namespaces_1122_after.png